src/wallet/signer.rs

```diff
@@ -521,10 +523,20 @@ impl PartialOrd for SignersContainerKey {
 
 impl Ord for SignersContainerKey {
     fn cmp(&self, other: &Self) -> Ordering {
-        self.ordering.cmp(&other.ordering)
+        self.ordering
+            .cmp(&other.ordering)
+            .then(self.id.cmp(&other.id))
+    }
+}
+
+impl PartialEq for SignersContainerKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.ordering == other.ordering
```

Comment:
Shouldn't this also compare the identifier?

Reply:
Makes sense, let me fix this.